c7n/filters/kms.py

```diff
@@ -68,17 +68,29 @@ def get_related_ids(self, resources):
         related_ids = super().get_related_ids(resources)
         normalized_ids = []
         for rid in related_ids:
+            if rid.startswith('alias'):
+                rid = self.alias_to_id.get(rid, rid)
             if rid.startswith('arn:'):
                 normalized_ids.append(rid.rsplit('/', 1)[-1])
             else:
                 normalized_ids.append(rid)
         return normalized_ids
 
     def process(self, resources, event=None):
+        self.alias_to_id = self.key_alias_to_key_id()
```

Comment:
Do we need this line here?

Reply:
We have to populate it somewhere. I initially thought we could "only do it once" by moving it outside the loop in `get_related_ids()`. But then [this](https://github.com/cloud-custodian/cloud-custodian/blob/c734cd08656cd332f3bac277693f08548bf2bfe6/c7n/filters/related.py#L59) caused `self.key_alias_to_key_id()` to get still get called once per resource.

This is breaking other related tests now, and I'm not sure if that means tests need to be reworked or the filter is still not right.

Reply:
Ah, now I see what you're saying

Reply:
When I tested re-recording some KMS tests, the original was spitting out a bunch of list alias calls. But I think what you were trying to do with reusing the already-populated alias map made sense :+1: .

I think we can keep that intent by building the alias lookup during `process` _and_ caching the response from `get_resource_manager()`. I also realized that if you're normalizing `get_related_ids()` to always use a key ID, we didn't need some of the alias lookup magic in `get_related()` (leaving it in would probably be misleading).